src/reference/torii-endpoints.md

```diff
@@ -1,69 +1,167 @@
+
+   ::: info
+
+   To learn more, see [Peer Configuration > Iroha Public Addresses](../guide/configure/peer-configuration.html#iroha-public-addresses).
+
+   :::
 
 ## API Version
 
-- **Protocol:** HTTP
-- **Method:** `GET`
-- **Endpoint:** `/api_version`
+- **Protocol**: `HTTP`
+- **Method**: `GET`
+- **Encoding**: `JSON`
+- **Endpoint**: `/api_version`
+
+#### Requests
 
-**Responses:**
+A `GET` request to the endpoint.
 
-`200 OK`: The current version of API used by Iroha returned as a JSON string. Grabbed from the genesis block's version,
-so at least a minimal subnet of 4 peers should be running and the genesis be submitted at the time of request.
+#### Responses
+
+| Code | Response              | Description                                                                   |
+| :--: | --------------------- | ----------------------------------------------------------------------------- |
+| 200  | OK                    | Returns the current version of the API used by Iroha 2.                       |
+| 401  | Unauthorized          | The client lacks valid credentials for the request.                           |
+| 500  | Internal Server Error | The server encountered an unexpected issue and could not fulfill the request. |
+
+**Example**:
 
 ```json
-"1"
+200 OK: "1"
 ```
 
+::: info
+
+The API version is retrieved from and is the same as the version of the [genesis block](../guide/configure/genesis.md), which means that at least a minimal subnet of four peers must be running, and the genesis block must already be submitted at the time of the request.
+
+:::
+
 ## Blocks Stream
 
-- **Protocol:** HTTP
-- **Protocol Upgrade:**`WebSocket
-- **Endpoint:** `/block/stream`
+- **Protocols**: `HTTP` upgraded to `WebSocket`
+- **Encoding**: `SCALE`
+- **Endpoint**: `/block/stream`
+
+#### Handshake
+
+Since the `/block/stream` endpoint handles continuous two-way data exchange, a `WebSocket` handshake between the client and server must first be performed to initiate communication with this endpoint.
 
-The client should send a [`BlockSubscriptionRequest`](/reference/data-model-schema#blocksubscriptionrequest) to initiate
-communication after the WebSocket handshake. Then the server sends a
-[`BlockMessage`](/reference/data-model-schema#blockmessage). Messages are SCALE-encoded[^1].
+The first HTTP request to this endpoint requires a standard set of `WebSocket` headers.
 
-Via this endpoint, the client first provides the starting block number (i.e. height) in the subscription request. After
-sending the confirmation message, the server starts streaming all the blocks from the given block number up to the
-current block and continues to stream blocks as they are added to the blockchain.
+**Example**:
+
+```http
+Host: example.com:8000
+Upgrade: websocket
+Connection: Upgrade
+Sec-WebSocket-Key: x3JJHMbDL1EzLkh9GBhXDw==
+Sec-WebSocket-Version: 13
+Sec-WebSocket-Extensions: permessage-deflate, client_max_window_bits
+```
```

Comment:
I am not sure we should cover details of how WebSocket works. It is pretty much a standard and API users might use any convenient WebSocket client they want. Such stuff about headers is performed under the hood and doesn't really matter for developers.

What matters and what is related to Iroha, however, is the flow of handshake and data exchange.